Wrong repository name in CI/CD

**Describe the bug**
When doing a pull request from a fork of the repository, the CI/CD will fail with the following error in the step `Install ByoDynaMo`.
```
bash: line 1: 404:: command not found
Error: Process completed with exit code 127.
```

This is because in `macos-system-ci.yml` and `ubuntu-system-ci.yml`, the branch name is taken from the user, for the name of the repository of the main. For example, in the PR #450:
```
2026-02-22T17:00:29.1565523Z env:
2026-02-22T17:00:29.1565762Z   GHA_BRANCH_NAME: fix/null-to-nullptr
2026-02-22T17:00:29.1566109Z   GHA_REPOSITORY: BioDynaMo/biodynamo
```

--- a/.github/workflows/macos-system-ci.yml
+++ b/.github/workflows/macos-system-ci.yml
@@ -38,7 +38,7 @@ jobs:
           echo "GHA_BRANCH_NAME=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
         fi
 
-        echo "GHA_REPOSITORY=${{ github.repository}}" >> $GITHUB_ENV
+        echo "GHA_REPOSITORY=${{ github.event.pull_request.head.repo.full_name || github.repository }}" >> $GITHUB_ENV
 
     - name: Install BioDynaMo
       shell: zsh {0}

--- a/.github/workflows/ubuntu-system-ci.yml
+++ b/.github/workflows/ubuntu-system-ci.yml
@@ -34,7 +34,7 @@ jobs:
           echo "GHA_BRANCH_NAME=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
         fi
 
-        echo "GHA_REPOSITORY=${{ github.repository }}" >> $GITHUB_ENV
+        echo "GHA_REPOSITORY=${{ github.event.pull_request.head.repo.full_name || github.repository }}" >> $GITHUB_ENV
 
     - name: Set OSVERS
       run: |

